### DISCUSSION TITLE
0.8.0-rc2 has been released!

### DISCUSSION THREAD

**Discussion by mihe** [godot-jolt/godot-jolt#616] [Announcements]
This is the second **release candidate** for version 0.8.0, and assuming no major bugs/regressions are found is likely to become 0.8.0-stable.

This release supports [**Godot 4.1**](https://github.com/godotengine/godot/releases/tag/4.1.1-stable).

## Changelog

These are the notable changes that have been made since [0.7.0-stable](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.7.0-stable) was released. You can also find a list of all the commits [here](https://github.com/godot-jolt/godot-jolt/compare/v0.7.0-stable...v0.8.0-rc2).

Any breaking changes are denoted with ⚠️.

### Changed

- ⚠️ Changed collision layers and masks for `Area3D` to behave like they do in Godot Physics, allowing for asymmetrical setups, where overlaps are only reported if the mask of an `Area3D` contains the layer of the overlapping object.
- ⚠️ Changed the `body_set_force_integration_callback` method of `PhysicsServer3D` to behave like it does with Godot Physics, where omitting the binding of `userdata` requires that the callback also doesn't take any `userdata`. It also will no longer be called when the body is sleeping.

### Added

- Added timings of Jolt's various jobs to the "Physics 3D" profiler category.
- Added registering of `JoltPhysicsServer3D` as an actual singleton, which makes Jolt-specific server methods (e.g. `pin_joint_get_applied_force`) easier to deal with from dynamic scripting languages like GDScript.
- Added `space_dump_debug_snapshot` to `JoltPhysicsServer3D`, for dumping a binary debug snapshot of a particular physics space.
- Added `dump_debug_snapshots` to `JoltPhysicsServer3D`, for dumping binary debug snapshots of all currently active physics spaces.
- Added a "Dump Debug Snapshots" menu option to "Project / Tools / Jolt Physics", for dumping binary debug snapshots of all the editor's physics spaces.

### Fixed

- Fixed issue with `move_and_slide`, where under certain conditions you could get stuck on internal edges of a `ConcavePolygonShape3D` if the floor was within 5-ish degrees of `floor_max_angle`.
- Fixed issue with `move_and_slide`, where under certain conditions, while using a `BoxShape3D` or `CylinderShape3D` shape, you could get stuck on internal edges of a `ConcavePolygonShape3D`.
- Fixed issue where collision with `ConvexPolygonShape3D` could yield a flipped contact normal.
- Fixed issue where an `Area3D` with `monitoring` disabled wouldn't emit any entered events for already overlapping bodies once `monitoring` was enabled.
- Fixed issue where changing the center-of-mass of a `RigidBody3D` attached to a joint would shift its transform relative to the joint.
- Fixed issue where the `total_gravity` property on `PhysicsDirectBodyState3D` would always return a zero vector for kinematic bodies.
- Fixed issue with `Area3D` detecting overlaps slightly outside of its collision shapes.

<hr /><em>This discussion was created from the release <a href='https://github.com/godot-jolt/godot-jolt/releases/tag/v0.8.0-rc2'>0.8.0-rc2</a>.</em>

**Comment by ZachAR3** [comment]
Awesome! Have you considered adding 2D physics once you get 3D finished? 2D physics (especially joints) is an absolute wreck right now, and considering how much better Jolt works than Godot in 3D I'm assuming it would be the same in 2D haha. Regardless, awesome work!

**Reply by mihe** [reply]
I have no such plans, no. There are however other 2D physics extensions in the works already, like the [Box2D](https://github.com/appsinacup/godot-box2d) one and the [Rapier2D](https://github.com/fabriceci/godot-rapier2d) one.

**Reply by ZachAR3** [reply]
Fair enough then. Thanks for the work on this!
